nautobot/extras/models/relationships.py

```diff
@@ -332,16 +332,26 @@ def clean(self):
             side_model = getattr(self, f"{side}_type").model_class()
             model_name = side_model._meta.label
             if not isinstance(filter, dict):
-                raise ValidationError(f"Filter for {model_name} must be a dictionary")
+                raise ValidationError({f"{side}_filter": f"Filter for {model_name} must be a dictionary"})
 
-            filterset = get_filterset_for_model(side_model)
-            if not filterset:
-                raise ValidationError(f"Filter are not supported for {model_name} object (Unable to find a FilterSet)")
+            filtersetclass = get_filterset_for_model(side_model)
+            if not filtersetclass:
+                raise ValidationError(
+                    {f"{side}_filter": f"Filter are not supported for {model_name} object (Unable to find a FilterSet)"}
+                )
+            filterset = filtersetclass(filter, side_model.objects.all())
+
+            if filterset.errors:
+                for key in filterset.errors:
+                    error_message = ", ".join(filterset.errors[key])
+                    raise ValidationError({f"{side}_filter": f"{key}: {error_message}"})
```

Comment:
Seems like an incorrect pattern to have a `for` loop that always raises an exception on the first item. Should we instead concatenate all of the `error_message` together before throwing a single combined `ValidationError`?

Reply:
I think you could just `raise ValidationError(filterset.errors)` since `filterset.errors` is a dict? Or is that not presenting the errors correctly?

Reply:
Good catch, I refactored the code to raise all errors at once, unfortunately I can't pass the dict directly so I constructed a list with all errors